### PR TITLE
docs: update guides to v5

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution
Solve #512 
- Complete update of guides, to includes all updates related to v5.
- Implemented in create account guide : creation of a Braavos account.
- Small fix in V4.22 doc (in signature guide).

## Usage related changes

Access to the V5 guides.

## Development related changes

N/A 

## Other 

These guides are made with the hypothesis that #633 is implemented.
For Braavos account creation, as a lot of new code is necessary, a module has been created and can be used by users. As it's outside of starknet.js code, `.eslintignore` and `tsconfig.json` have been updated accordingly.

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] All tests are passing

### CODE REVIEW

File: www/docs/guides/interact.md [hunk 1]
```diff
@@ -83,77 +83,69 @@ const myTestContract = new Contract(testAbi, testAddress, provider);
...
+await provider.waitForTransaction(result.transaction_hash);
 ```
+
+## Other existing methods
+
+Some other useful method to interact with Starknet :
+
+### Function name defined in the code
+
+If you want to call a function with its name caontained in a variable :
```
Comment (ivpavici) [inline www/docs/guides/interact.md:130]:
```suggestion
If you want to call a function with its name contained in a variable :
```

File: www/docs/guides/interact.md [hunk 2]
```diff
@@ -83,77 +83,69 @@ const myTestContract = new Contract(testAbi, testAddress, provider);
...
+
+```typescript
+const listFn = ["calc-sum","calc-hash","calc-proof"];
+// fnChoice is a number defined during execution
+const res = await myTestContract[listFn[fnChoice]](200, 234567897n, 865423);
+```
+
+### Light and fast call
+
+If you want to have a very fast execution, with the minimum of ressource usage :
```
Comment (ivpavici) [inline www/docs/guides/interact.md:140]:
```suggestion
If you want to have a very fast execution, with the minimum of resource usage :
```

File: www/docs/guides/what_s_starknet.js.md [hunk 3]
```diff
@@ -23,6 +23,8 @@ Some important topics that have to be understood:
   - private customized version of Starknet.
   - local Starknet node (connected to mainnet or testnet).
 
+> Understand what is Starknet and how it works is necessary. Then, you can learn how to interact with it using Starknet.js. So, at this stage, you should be aware of the content of the [Starknet official doc](https://docs.starknet.io/documentation/)
```
Comment (ivpavici) [inline www/docs/guides/what_s_starknet.js.md:26]: we could add also https://github.com/starknet-edu/starknetbook

File: .eslintignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 node_modules/
 dist/
+www/
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply .eslintignore:3]: Explained in PR description : 
For Braavos account creation, as a lot of new code is necessary, a module has been created and can be used by users. As it's outside of starknet.js code, .eslintignore and tsconfig.json have been updated accordingly.

[the remark this answers is not in the document]
Comment (ivpavici) [reply .eslintignore:3]: ah indeed! sorry!

File: www/docs/guides/connect_network.md [hunk 5]
```diff
@@ -76,12 +82,19 @@ For example, if you want to read the events recorded in a range of blocks, you n
 ```typescript
 import { RpcProvider } from "starknet";
 const providerRPC = new RpcProvider({ nodeUrl: "http://192.168.1.99:9545" }); // for a pathfinder node located in a PC in the local network
+const lastBlock=await providerRPC.getBlock('latest');
 let eventsList = await providerRPC.getEvents({
     address: myContractAddress,
-    from_block: {block_number: block_number0},
-    to_block: {block_number: block_number1},
-    chunk_size: 1000
+    from_block: {block_number: lastBlock.block_number-2},
```
[the remark this answers is not in the document]
Comment (PhilippeR26) [reply www/docs/guides/connect_network.md:88]: The target is to search in the last 3 blocks.

File: www/docs/guides/what_s_starknet.js.md (hunk 3 above)
Comment (PhilippeR26) [reply www/docs/guides/what_s_starknet.js.md]: Implemented

File: www/docs/guides/connect_network.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ivpavici) [reply www/docs/guides/connect_network.md:88]: is that - like minus 2  ? :D

**Review by ivpavici — APPROVED** [review]

File: www/docs/guides/compiled_contracts/deployBraavos.ts [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  Provider,
+  RawCalldata,
+  constants,
+  ec,
+  hash,
+  num,
+  stark,
+} from 'starknet';
+
+const BraavosProxyClassHash: BigNumberish =
```
Comment (penovicp) [inline www/docs/guides/compiled_contracts/deployBraavos.ts:23]: Added a commit that changes some type references, e.g. here the change was from `num.BigNumberish` to `BigNumberish`. `num.BigNumberish` will still be supported but direct imports or using it as `types.BigNumberish` should be preferred. This was a change introduced in PR #627 to make the type usage consistent across the library.

**Review by penovicp — APPROVED** [review]

**Pull request merged**